### PR TITLE
Modifies the welding fuel tank sprites on april fools

### PR DESCRIPTION
**Pull request by Cyberboss**
A well thought out suggestion with absolutely no downsides.

### CODE REVIEW

**Comment by Farquaar** [comment]
Does it make the iconic kaplunk sound when it explodes?

File: code/modules/reagents/reagent_dispenser.dm [hunk 1]
```diff
@@ -86,6 +86,12 @@
 	icon_state = "fuel"
 	reagent_id = /datum/reagent/fuel
 
+/obj/structure/reagent_dispensers/fueltank/Initialize(mapload)
+	. = ..()
+
+	if(SSevents.holidays && SSevents.holidays[APRIL_FOOLS])
+		icon_state = "fuel_fools"
```
Comment (LemonInTheDark) [inline code/modules/reagents/reagent_dispenser.dm:93]: ~~can we color it based off the contained reagent~~

**Comment by ATH1909** [comment]
ooh, that reminds me, we could switch the sprites for ignited and unignited welders for april fools as well

**Comment by BurgerLUA** [comment]
give me credit at least smh

**Comment by BurgerLUA** [comment]
image
Might as well do the water one while you're at it.

**Comment by Cyberboss** [comment]
DM me a .dmi :honk:

**Comment by BurgerLUA** [comment]
> DM me a .dmi :honk:

You could just copy and paste the bottom barrel sprite into dream maker and it will work.

**Comment by LemonInTheDark** [comment]
ready?

**Comment by BurgerLUA** [comment]
> ready?

Correct, up to date sprites are used and there seems to be no code issues.

(Yes I know I'm not the OP but I really want this merged)

File: code/modules/reagents/reagent_dispenser.dm [hunk 2]
```diff
@@ -16,6 +16,12 @@
 	///Is this source self-replenishing?
 	var/refilling = FALSE
 
+/obj/structure/reagent_dispensers/Initialize(mapload)
+	. = ..()
+
+	if(icon_state == "water" && SSevents.holidays && SSevents.holidays[APRIL_FOOLS])
```
Comment (MrMelbert) [inline code/modules/reagents/reagent_dispenser.dm:22]: why do you sanity check for `SSevents.holidays` on one but not the other

**Review by Twaticus — APPROVED** [review]

File: code/modules/reagents/reagent_dispenser.dm [hunk 3]
```diff
@@ -86,6 +92,12 @@
 	icon_state = "fuel"
 	reagent_id = /datum/reagent/fuel
 
+/obj/structure/reagent_dispensers/fueltank/Initialize(mapload)
+	. = ..()
+
+	if(SSevents.holidays && SSevents.holidays[APRIL_FOOLS])
```
Comment (ShizCalev) [inline code/modules/reagents/reagent_dispenser.dm:98]: if(SSevents?.holidays[APRIL_FOOLS])

Comment (Mothblocks) [reply code/modules/reagents/reagent_dispenser.dm]: typo, `if(SSevents.holidays?[APRIL_FOOLS])`

File: code/modules/reagents/reagent_dispenser.dm [hunk 4]
```diff
@@ -16,6 +16,12 @@
 	///Is this source self-replenishing?
 	var/refilling = FALSE
 
+/obj/structure/reagent_dispensers/Initialize(mapload)
+	. = ..()
+
+	if(icon_state == "water" && SSevents.holidays?[APRIL_FOOLS])
```
Comment (LemonInTheDark) [inline code/modules/reagents/reagent_dispenser.dm:22]: ~~Leaves should never be the same as branches~~

**Comment by optimumtact** [comment]
This is not appropriate for master and should have been a test merge only

**Comment by LemonInTheDark** [comment]
I thought it was funny I'm sorry

**Pull request merged**